spec.html

```diff
@@ -28538,135 +28360,96 @@ <h1>Math.imul ( _x_, _y_ )</h1>
 
       <emu-clause id="sec-math.log">
         <h1>Math.log ( _x_ )</h1>
-        <p>Returns an implementation-approximated value representing the result of the natural logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0* or *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <p>Returns the natural logarithm of _x_.</p>
+        <p>When the `Math.log` method is called with argument _x_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ is *+&infin;*, return _n_.
+          1. If _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
```

Comment:
Better to go in descending order:
```suggestion
          1. If _n_ is 1, return *+0*.
          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
          1. If _n_ &lt; 0, return *NaN*.
```

Reply:
I had organized the steps to list all "identity" cases, followed by exceptional cases (e.g. inputs which return NaN or that have specific return values) and then the "regular" cases. 

How do you want to handle cases like `Math.atanh`?

```
          1. Let _n_ be ? ToNumber(_x_).
          1. If _n_ is *NaN*, _n_ is *+0*, or _n_ is *-0*, return _n_.
          1. If _n_ &gt; 1 or _n_ &lt; -1, return *NaN*.
          1. If _n_ is +1, return *+&infin;*.
          1. If _n_ is -1, return *-&infin;*.
          1. Return an implementation-approximated value representing the result of the inverse hyperbolic tangent of _n_.
```

To strictly follow your guidance, I could split the third step so that the order is > 1, = 1, = -1, < -1. Alternatively this could remain as is and we treat multiple conditions as special cases that precede the descending order. Probably other options too. What do you think, @bakkot?

Reply:
@bakkot See https://github.com/tc39/ecma262/pull/2122#discussion_r465876073.

Reply:
> How do you want to handle cases like `Math.atanh`?

In cases where there's no natural order because some conditions are combined this preference wouldn't apply. I only intended this comment to apply to the `log` methods, and only for these three cases (1, 0, <0).

@michaelficarra I saw that comment, but I don't understand the relevance here.

Reply:
@bakkot - Thanks! I've made a few related changes to prefer descending order (e.g. [this one](https://github.com/tc39/ecma262/pull/2122/commits/e7c17df9c70b0575e0d483fc09b67fbeb50a452d#diff-3540caefa502006d8a33cb1385720803R28309)) but the `log` entries were indeed the primary ones.